runners/direct-java/src/main/java/org/apache/beam/runners/direct/ParDoEvaluatorFactory.java

```diff
@@ -104,12 +104,12 @@ public void cleanup() throws Exception {
             .getExecutionContext(application, inputBundle.getKey())
             .getOrCreateStepContext(stepName, stepName);
 
-    DoFnLifecycleManager fnManager =
-        fnClones.getUnchecked(hooks.getDoFn(application.getTransform()));
+    DoFn<InputT, OutputT> doFn = hooks.getDoFn(application.getTransform());
+    DoFnLifecycleManager fnManager = fnClones.getUnchecked(doFn);
     try {
       return DoFnLifecycleManagerRemovingTransformEvaluator.wrapping(
           hooks.createParDoEvaluator(
-              evaluationContext, application, stepContext, (DoFn<InputT, OutputT>) fnManager.get()),
+              evaluationContext, application, stepContext, doFn),
```

Comment:
ah, yes. Is there a way to avoid passing in `DoFn` ? I find eventually `SimpleDoFnRunner` would need it to extend `DoFn.Context` although it's not used in other places.